packages/library-base/src/commonMain/kotlin/io/realm/TypedRealm.kt

```diff
@@ -10,14 +10,31 @@ interface TypedRealm : BaseRealm {
     /**
      * Returns the results of querying for all objects of a specific type.
      *
-     * For a [Realm] instance this reflects the state of the realm at the invocation time, thus
+     * For a [Realm] instance this reflects the state of the Realm at the invocation time, thus
      * the results will not change on updates to the Realm. For a [MutableRealm] the result is live
      * and will in fact reflect updates to the [MutableRealm].
      *
      * @param clazz The class of the objects to query for.
      * @return The result of the query.
      */
     open fun <T : RealmObject> objects(clazz: KClass<T>): RealmResults<T>
+
+    /**
+     * Returns a [RealmQuery] matching the predicate represented by [query].
+     *
+     * For a [Realm] instance this reflects the state of the Realm at the invocation time, this
+     * the results obtained from the query will not change on updates to the Realm. For a
+     * [MutableRealm] the query will produce live results and will in fact reflect updates to the
+     * [MutableRealm].
+     *
+     * @param query the Realm Query Language predicate to append.
+     * @param args Realm values for the predicate.
+     */
+    fun <T : RealmObject> query(
+        clazz: KClass<T>,
+        query: String = "TRUEPREDICATE",
```

Comment:
How does this look in the IDE...is it visible in the type hints that this is the default when using a normal Realm? If not, I would suggest moving this default value to the `Realm` and `MutableRealm` classes, even though it means we are duplicating it.